Source/Adapters/Blackjack.Adapters/UserAdapter.cs

```diff
@@ -47,32 +54,37 @@ public UserAdapter(IUserFacade userFacade)
     public async Task CreateUser(string username)
     {
         string soundex = CreateSoundex(username);
-        UserBE userBySoundex = await GetUsers(username);
-        if (userBySoundex != null)
+        UserBE userBE = await GetUserBySoundex(username);
```

Comment:
You're not passing a soundex code to this method you're passing a username.

What that method does w/ that username is its business and nobody else's business.

This is not GetUserBySoundex. This is GetUserByUsername.